src/main/java/training/subsystems/Elbow/ElbowSubsystem.java

```diff
@@ -0,0 +1,76 @@
+package training.subsystems.Elbow;
+
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkPIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import utils.GBSubsystem;
+
+import static training.subsystems.Elbow.Constants.*;
+
+public class ElbowSubsystem extends GBSubsystem {
+    private static ElbowSubsystem instance;
+
+    private final CANSparkMax motor = new CANSparkMax(ELBOW_ID, CANSparkLowLevel.MotorType.kBrushless);
+    private final SparkPIDController motorPIDF = motor.getPIDController();
+    private Rotation2d angle = DEFAULT_POSITION_ELBOW;
+
+    private ElbowSubsystem() {
+        motorPIDF.setP(ELBOW_PID_CONTROLLER.getP());
+        motorPIDF.setD(ELBOW_PID_CONTROLLER.getD());
+        motorPIDF.setI(ELBOW_PID_CONTROLLER.getI());
+    }
+
+    public static ElbowSubsystem getInstance() {
+        if (instance == null) {
+            instance = new ElbowSubsystem();
+        }
+        return instance;
+    }
+
+    public void moveWritst(double power) {
+        if (Math.abs(power) < POWER_LIMIT_ELBOW) {
+            motor.set(power);
+        }
+    }
+
+    public Rotation2d getMotorAngle() {
+        return Rotation2d.fromRotations(motor.getEncoder().getPosition());
+    }
+
+    public Rotation2d getDesiredAngle() {
+        return angle;
+    }
+
+    public void setAngle(Rotation2d angle) {
+        this.angle = angle;
+    }
+
+    public void addToAngle(Rotation2d angle) {
+        this.angle = this.angle.plus(angle);
+    }
+
+    public void subtractFromAngle(Rotation2d angle) {
+        this.angle = this.angle.minus(angle);
+    }
+
+    public boolean isAtAngle(Rotation2d angle) {
+        double anglesDelta = (this.angle.getDegrees() % 1) - (angle.getRadians() % 1);
+        return Math.abs(anglesDelta) <= ELBOW_TOLERANCE;
+    }
+
+    @Override
+    protected String getLogPath() {
+        return ELBOW_LOGPATH;
+    }
+
+    @Override
+    public void subsystemPeriodic() {
+        motorPIDF.setReference(
+                angle.getRotations() / ELBOW_GEAR_RATIO % 1,
+                CANSparkBase.ControlType.kPosition, 0,
+                ELBOW_FEEDFORWARD.calculate(0, 0)
```

Comment:
```suggestion
                ELBOW_FEEDFORWARD.calculate(targetAngleWhenAngleEquals0WhenArmIsParallelToGround, 0)
```

Reply:
FF doesn't need a target angle. Its value changed by the first and second derivatives. I changed the first param to encoder.getVolocity because it impacts the sign of kS, but does not change the length of the vector. I don't want kV and kA to take effect since I use FF mostly to resist gravity and friction. As a result, the first param just needs to be 1 or -1 depending on the direction. See this page. correct me if I'm wrong. sgn is defined as the sign of the func, and dots are Newton's sign for derivatives (instead of ') which is sometimes being used in the context of physics.

Reply:
OK I should used elevatorFF instead of armFF